app/controllers/events_controller.rb

```diff
@@ -0,0 +1,36 @@
+class EventsController < ApplicationController
+  before_action :check_user, only: %i[new create]
+  def index
+    @events = Event.all
```

Comment:
- [ ] Send @upcoming_events and @past_events from here instead of accessing them in the view.